apps/examples/testcase/le_tc/filesystem/itc_fs.c

```diff
@@ -0,0 +1,1442 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_fs.c
+/// @brief Scenarios for Filesystem
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <dirent.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/mount.h>
+#include <sys/stat.h>
+#include "tc_common.h"
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_FS_SMARTFS
+
+#if defined(CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS)
+#define MOUNT_DEV_DIR CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS_DEVNAME
+
+#elif defined(CONFIG_ARTIK05X_AUTOMOUNT_USERFS)
+#define MOUNT_DEV_DIR CONFIG_ARTIK05X_AUTOMOUNT_USERFS_DEVNAME
+
+#else
+#define MOUNT_DEV_DIR "/dev/smart1"
+#endif
+
+#define TARGET_FS_NAME "smartfs"
+#else
+#define MOUNT_DEV_DIR "/dev/fs1"
+#define TARGET_FS_NAME "unknwon"
+#endif
+
+#define BUFFER_LEN 30
+#define MOUNT_DIR CONFIG_MOUNT_POINT
+#define VFS_FILE_PATH MOUNT_DIR"vfs"
+#define VFS_FOLDER_PATH MOUNT_DIR"folder"
+#define VFS_LOOP_COUNT 5
+#define VFS_TEST_CONTENTS_1 "THIS IS VFS TEST 1"
+#define VFS_TEST_CONTENTS_2 "THIS IS VFS TEST 2"
+
+/**
+* @fn               add_dirs
+* @description      add directory VFS_FOLDER_PATH, and count of inner directories
+* @return           void
+*/
+static void add_dirs(int count)
+{
+	char folder_name[BUFFER_LEN];
+	int len;
+	int i;
+	strcpy(folder_name, VFS_FOLDER_PATH);
+	strcat(folder_name, "/0\0");
+	len = strlen(folder_name);
+	mkdir(VFS_FOLDER_PATH, 0777);
+	for (i = 0; i < count; i++) {
+		folder_name[len - 1] = '0' + i;

```

Comment:
How about make function like 'generate_childdir(char* dir, int num)' add_dirs & remove_dirs can use it.

Reply:
also there's itoa in TizenRT